### PR TITLE
Refactor PerformanceController

### PR DESCRIPTION
**Pull request by kenzieschmoll**
This PR contains a large refactor to split the PerformanceController up into more manageable pieces. The PerformanceController class previously contained all business for most of the features on the page. After this PR, that business / controller logic is split up into multiple "feature controllers":
- `FlutterFramesController`: handles all the business logic related to the flutter frames chart
- `TimelineEventsController`: handles all the business logic related to the timeline events tab (legacy and new perfetto impls)
- `RasterStatsController`: handles all the business logic related to the raster stats tab (this controller already existed, but I moved the remaining raster-stats-related logic from PerformanceController over to RasterStatsController.

Along with breaking up the controller code, I broke up the tests as well. This resulted in an empty file for `performance_controller_test.dart`, but I left it with a TODO to add better test coverage for the remaining login in `PerformanceController`. 

There should not be any functional changes in this PR - all just slicing and dicing and moving stuff around.

### CODE REVIEW

File: packages/devtools_app/test/performance/enhance_tracing_test.dart [hunk 1]
```diff
@@ -2,12 +2,8 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-import 'package:devtools_app/src/config_specific/ide_theme/ide_theme.dart';
+import 'package:devtools_app/devtools_app.dart';
```
Comment (kenzieschmoll) [inline packages/devtools_app/test/performance/enhance_tracing_test.dart:5]: per a previous [discussion](https://github.com/flutter/devtools/pull/4566#discussion_r999849325) on another PR, it is acceptable to import devtools_app.dart into test files.

**Review by CoderDake — APPROVED** [review]
Kenzie took me on a walkthrough of the PR. She answered my questions about her confidence in the changes. She was able to reasure me that our tests would be failing if the critical paths weren't working any more, and walk me through the main code movements.

I'm not able to really approve this from my own understasnding of the diff, but the chat with Kenzie has me confident enough to trust her intuitions on the change

File: packages/devtools_app/lib/src/screens/performance/panes/flutter_frames/flutter_frames_controller.dart [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+  ValueListenable<bool> get showFlutterFramesChart => _showFlutterFramesChart;
+  final _showFlutterFramesChart = ValueNotifier<bool>(true);
+  void toggleShowFlutterFrames(bool value) =>
+      _showFlutterFramesChart.value = value;
+
+  /// Whether flutter frames are currently being recorded.
+  ValueListenable<bool> get recordingFrames => _recordingFrames;
+  final _recordingFrames = ValueNotifier<bool>(true);
+
+  ValueListenable<bool> get badgeTabForJankyFrames => _badgeTabForJankyFrames;
```
Comment (bkonyi) [inline packages/devtools_app/lib/src/screens/performance/panes/flutter_frames/flutter_frames_controller.dart:40]: Nit: While we're here, maybe add doc comments for properties that don't have them?

Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/screens/performance/panes/flutter_frames/flutter_frames_controller.dart:40]: good idea. Will do this as a follow up.

**Pull request merged**